kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -184,15 +184,17 @@ object Abort:
                     accept = [C] =>
                         input =>
                             input.isPanic ||
-                                input.asInstanceOf[Error[Any]].failure.exists {
-                                    case e: E => true
-                                    case _    => false
-                            },
+                                input.asInstanceOf[Error[Any]].failure.exists(ct.accepts),
                     handle = [C] => (input, _) => input,
                     recover =
-                        case fail: E if classOf[Throwable].isAssignableFrom(ct.runtimeClass) =>
+                        case ct(fail) if ct <:< SafeClassTag[Throwable] =>
                             Result.fail(fail)
-                        case fail => Result.panic(fail)
+                        case fail =>
+                            Result.panic(fail),
+                    done =
+                        case Result.Panic(ct(e)) if ct <:< SafeClassTag[Throwable] =>
+                            Result.Fail(e)
```

Comment:
I noticed a scenario when working on tests where an exception type that was considered a panic in an inner scope continued as a panic even if an outer scope was handling it, which seems an inconsistent state. This is why some tests have changes to match against `Result.fail` instead of `Result.panic`.

Reply:
I am not sure I agree with this change. I would have expected that you would need to explicitly lift a Panic -> Fail, like ZIO's resurrect. If you use `Abort.run[Throwable]`, will this lift all Panics to Fail?

Reply:
Yes, it'd transform panics. It felt more natural for `Abort.run[Throwable]` to "catch" any failures in a computation, even if panics. It seems better to discuss that separately, though. I'll revert this change.